docs/graphcast/radios/intro.md

```diff
@@ -31,251 +28,196 @@ Once that's done, you can start building your very first Radio.
 
 ### Populate your `.env` file
 
-You now need to specify five essential variables in your `.env` file, here's the `.env.example`:
-
-```bash
-ETH_NODE=
-RADIO_OPERATOR_PRIVATE_KEY=
-NETWORK_SUBGRAPH=
-INDEXER_MANAGEMENT_SERVER=
-GRAPH_NODE=
-```
+You now need to export two environment variables, `ETH_NODE` (url to an Ethereum node) and `PRIVATE_KEY` (your operator address private key).
 
 Please use a Goerli ETH node for this example (doesn't need to be a full node).
 
 ### A few dependencies
 
-For this example you need to have the `@graphops/graphcast`, `dotenv` and `@graphprotocol/common-ts` npm packages installed.
-
-### The imports
-
-Create a new file, for instance we'll be using `index.ts`, and import the following packages:
+Start off with a new Rust project (`cargo new ping-pong`). Then add the following dependencies to you `Cargo.toml` file:
 
-```typescript
-import "dotenv/config";
-import { ClientManager, GossipAgent } from "@graphops";
-import { createLogger } from "@graphprotocol/common-ts";
 ```
-
-:::tip
-If you need a refresher on what `ClientManager` and `GossipAgent` do, refer to [this earlier section of the Graphcast docs](http://docs.graphops.xyz/graphcast/sdk#gossip-agent).
-:::
-
-### Set up a few constants
-
-Add these constants right after the imports:
-
-```typescript
-const RADIO_PAYLOAD_TYPES = [{ name: "content", type: "string" }];
-const DOMAIN = "ping-pong";
-
-const logger = createLogger({
-  name: DOMAIN,
-  async: false,
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  level: process.env.logLevel as any,
-});
+[dependencies]
+graphcast_sdk = "0.0.1"
+once_cell = "1.15"
+tokio = { version = "1.1.1", features = ["full"] }
+anyhow = "1.0.39"
+ethers = "1.0.0"
 ```
 
-`RADIO_PAYLOAD_TYPES` defines the types for our Radio-specific message payload.
-`DOMAIN` is the unique name of the Radio, which is used in many places, but mainly to construct message topics.
+### The imports
 
-:::info
-Topics in the Graphcast network use the following format - `/graphcast/{version}/{domain}/{subtopic}/proto}`. In our ping pong example we won't need a `subtopic`, which is an optional field where the topics need to branch out (as is the case in the POI cross-checker radio, where a subgraph hash is provided as a `subtopic`, because rarely an Indexer would want to listen for POIs for **all** the subgraphs). But in the current case our topic will look like this - `/graphcast/0/ping-pong/proto`.
-:::
+Open your `main.rs` file and add the following imports:
 
-`logger` will be used to structure logs in a better way than the good ol' `console.log` (very important if we want to have better tracing and metrics in the future).
+```Rust
+use ethers::{
+    providers::{Http, Middleware, Provider},
+    types::U64,
+};
+use graphcast_sdk::gossip_agent::message_typing::GraphcastMessage;
+use graphcast_sdk::gossip_agent::GossipAgent;
+use once_cell::sync::OnceCell;
```

Comment:
I think it would be nice to add comments here to explain what each additional dependency is for, or it might be better to add comments in the actual example code. However I do think these are relatively straightforward so it's alright to leave as is

Reply:
good suggestion!